web/pdf_viewer.js

```diff
@@ -2256,13 +2293,49 @@ class PDFViewer {
     if (!this.pdfDocument) {
       return;
     }
-    this.#annotationEditorMode = mode;
-    this.eventBus.dispatch("annotationeditormodechanged", {
-      source: this,
-      mode,
-    });
 
-    this.#annotationEditorUIManager.updateMode(mode, editId, isFromKeyboard);
+    if (
+      mode === AnnotationEditorType.NONE ||
+      this.#annotationEditorMode === AnnotationEditorType.NONE
+    ) {
```

Comment:
Given this condition it seems, unless I'm misunderstanding things, that this method now only supports switching to/from editing but not switching *directly* between two different editing modes (e.g. from Ink to FreeText)?
Hence in that case I don't see that we ever dispatch the events or update the mode, and testing in the viewer it doesn't seem to work.

Also, given that enabling of editing could be delayed since it's now async (by waiting for "pagerendered"), what happens if there's a new editing switching before a previous one is complete?
Since, as far as I can tell, there's nothing "aborting" a previous delayed switching so is there not a risk that the state could become wrong?